web/components/admin/Toolbar/index.js

```diff
@@ -189,6 +221,8 @@ Toolbar.propTypes = {
   pageType: PropTypes.string,
   // Exports all applicants into a csv
   exportApplicants: PropTypes.func,
+  // fuzzy search on applicant name and email
+  searchApplicants: PropTypes.func,
 };
 
-export default Toolbar;
+export default withFirebase(Toolbar);
```

Comment:
For part B, instead of writing a cloud function, you can use react-redux-firebase to automatically listen to the `nfc_devices` collection. You can do that by replacing the export for the toolbar with `export default compose(firestoreConnect(['nfc_devices']))(Toolbar);`. Then to fetch the data, you can do something like `const { firestore: { data: { nfc_devices : devices } } } = this.props` and then devices will contain the `nfc_devices` documents. 

You can see an example of this in `web\components\admin\FilterModal\index.js` (where it listens to the `tags` collection)

Reply:
Fixed! Had no idea I could do it this way. Thanks!!